numpy/lib/function_base.py

```diff
@@ -52,6 +53,107 @@
     ]
 
 
+class QuantileInterpolation(Enum):
```

Comment:
I think it would be helpful if it remains public.
If someone want to extend the enum with it's own alpha and beta because none of the existing method fit his/her needs, having the enum public would allow that.

Reply:
Personally I'd be okay with its addition.
Nevertheless, this could definetely use some further discussion, especially on the namespaces where we'd want to expose it (`np`, `np.lib`, `np.lib.function_base`?).

Reply:
We discussed this a bit in the triage call, and I think the gist of it was: Do we really want to have both strings and the enum?  And if we have to pick, at least I still tend towards preferring strings.  (Which right now is also the backcompatible method).

(We did not have a very strong opinion on this though)

About using `interpolation=` or `method=`, I still think I prefer method. But there is a good point that it makes a good followup.  I have to take some quality time on the rest of the PR, before I want to do a final bike-shedding on the names.

Reply:
```suggestion
class _QuantileInterpolation(Enum):
```
Continuing from https://github.com/numpy/numpy/pull/19857#discussion_r714214190:
It would be best for now to make the enum private (and remove all mention of it from the release notes & public API docs).